### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by NasyTheSlayer**
[DEMO LINK](https://NasyTheSlayer.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +17,105 @@ export const goodsFromServer = [
...
+  const [sortField, setSortField] = useState(SortType.clear);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    isReverse,
+  );
+
+  const handleReverse = () => setIsReverse(!isReverse);
```
Comment (anastasiia-tilikina) [inline src/App.tsx:63]: I believe this halper is redundant,  you can call setIsReverse directly onClick.
Also when you set new state based on the previous one always use callback:

```typescript
setIsReverse(curr => !curr);
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +17,105 @@ export const goodsFromServer = [
...
-        >
-          Reset
-        </button>
+        {(isReverse || sortField)
+        && (
+          <button
+            onClick={() => {
+              setSortField(SortType.clear);
+              setIsReverse(false);
+            }}
```
Comment (anastasiia-tilikina) [inline src/App.tsx:104]: It's better to move this one to a separate function, though

**Review by anastasiia-tilikina — CHANGES_REQUESTED** [review]
Well done!
Let's just make a couple of changes in your code

**Review by VitaliyBodnar — APPROVED** [review]
Good job!

**Pull request closed without merging**